app/src/main/java/com/kamron/pogoiv/MainActivity.java

```diff
@@ -106,6 +106,7 @@ public void onReceive(Context context, Intent intent) {
     private final BroadcastReceiver showUpdateDialog = new BroadcastReceiver() {
         @Override
         public void onReceive(Context context, Intent intent) {
+            assert BuildConfig.isInternetAvailable;
```

Comment:
Do those assertions run in release builds? I've never used them but typically the jvm runs with them disabled by default, no idea at all for Android. 

Oh wow yeah that's really unnecessary for the `CrashlyticsWrapper`

It's not doing any real harm, just not necessary and maybe adding some code to the offline build that isn't ever used. There's something to be said for not having too many layers of abstraction so it might be fine in the settings activity.

Maybe it's worth a more robust internet check? This one is just "am I in the online flavour?". A better check might be to see if we have the internet permission (so it isn't relying on something in the build configuration) or maybe checking if you have internet _available_ when it's checked. The latter might require another permission, can't remember off the top of my head